File: 2798.py
numOfCards, maxSum = map(int, input().split())
cards = list(map(int, input().split()))
bestRecord = 0

for i in range(0, numOfCards) :
    for j in range(i + 1, numOfCards) :
        for k in range(j + 1, numOfCards):
            sumOfCards = cards[k] + cards[j] + cards[i]
            if sumOfCards == maxSum :
                bestRecord = sumOfCards
                break;
            elif sumOfCards > bestRecord and sumOfCards < maxSum :
                bestRecord = sumOfCards
                
print(bestRecord)

File: 2798.c
#include<stdio.h>

int main(){
    int numOfCards, maxSum;
    int bestRecord = 0;
    int cards[100] = {0, };
    
    scanf("%d %d", &numOfCards, &maxSum);
    for(int i = 0; i < numOfCards; i++)
        scanf("%d", &cards[i]);
    
    for(int i = 0; i < numOfCards; i++){
        for(int j = i + 1; j < numOfCards; j++){
            for(int k = j + 1; k < numOfCards; k++){
                if(maxSum == cards[k] + cards[j] + cards[i]){
                    bestRecord = maxSum;
                    break;
                }
                else if(maxSum > cards[k] + cards[j] + cards[i] && cards[k] + cards[j] + cards[i] > bestRecord)
                    bestRecord = cards[k] + cards[j] + cards[i];
            }
        }
    }
    printf("%d", bestRecord);
    return 0;
}
